src/support/ios.c

```diff
@@ -821,7 +821,8 @@ size_t ios_copyuntil(ios_t *to, ios_t *from, char delim)
         }
         else {
             size_t ntowrite = pd - (from->buf+from->bpos) + 1;
-            written = ios_write(to, from->buf+from->bpos, ntowrite);
+            size_t nchomp = ios_nchomp(from, ntowrite, chomp);
```

Comment:
I'd check `chomp` here, and not pass it at all to `ios_nchomp`. That can avoid a function call when `chomp` is false. Then you can assume `chomp` is true inside `ios_nchomp`.